hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -4565,17 +4663,23 @@ private void doMiniBatchMutate(BatchOperation<?> batchOp) throws IOException {
       lock(this.updatesLock.readLock(), miniBatchOp.getReadyToWriteCount());
       locked = true;
 
+      // From this point until memstore update this operation should not be interrupted.
+      disableInterrupts();
```

Comment:
I think here we could have a checkInterrupt? And for me, I think before STEP 4, we could always interrupt the handler but I wonder whether it worth. The prepare mini batch and build WAL are all in memory operations so they should be fast. So I suggest that, we call checkInterrupt here, if we pass, then remove the handler from the Map.

Reply:
Yes, here I can put a checkInterrupt. Let me do that.

If we remove the handler from the map at this point, no other mini batch will be eligible for interrupt. 

batchMutate does this:

    while (!batchOp.isDone()) {
        ...
        doMiniBatchMutate(batchOp);
        ...
      }

If we remove the handler from the map only the first iteration through the loop can be interrupted.

Reply:
Oh, you are right, I recall that we have a loop and will only acquire row locks as many as possible in each loop. Then this should be fine.